crates/storage/provider/src/traits/chain.rs

```diff
@@ -0,0 +1,140 @@
+///! Canonical chain state notification trait and types.
+use crate::PostState;
+use auto_impl::auto_impl;
+use core::fmt;
+use reth_primitives::{BlockNumHash, BlockNumber, Receipt, SealedBlockWithSenders, TxHash};
+use std::{collections::BTreeMap, fmt::Formatter, sync::Arc};
+use tokio::sync::broadcast::{Receiver, Sender};
+
+/// Trait that holds blocks and post state of execution those blocks.
```

Comment:
I see, so this basically represents a range of executed blocks with the `PostState` as the result